spyder/api/plugins/new_api.py

```diff
@@ -1,228 +1,398 @@
+    Configuration options added by this plugin for other plugins.
+
+    Examples
+    --------
+
+    .. code-block:: python
+
+        ADDITIONAL_CONF_OPTIONS = {
+            "section": {
+                "option_1": True,
+                "option_2": "default_value",
+            },
+        }
+    """
+
+    ADDITIONAL_CONF_TABS: (
+        dict[str, list[type[SpyderPreferencesTab]]] | None
+    ) = None
+    """
+    Define additional tabs of option for other plugins' configuration pages.
+
+    All configuration tabs should inherit from
+    :class:`~spyder.api.preferences.SpyderPreferencesTab`.
+
+    Examples
+    --------
+
+    .. code-block:: python
+
+        ADDITIONAL_CONF_TABS = {"plugin_name": [MyPluginsSpyderPrefsTab]}
+    """
+
+    CUSTOM_LAYOUTS: list[
+        type[spyder.plugins.layout.api.BaseGridLayoutType]
+    ] = []
+    """
+    Define custom layout classes that the plugin wants to be registered.
+
+    The custom classes should extend from
+    :class:`spyder.plugins.layout.api.BaseGridLayoutType`.
+    """
+
+    IMG_PATH: str | None = None
+    """
+    Path for images relative to the plugin path.
+
+    A Python package can include one or several Spyder plugins. If the latter,
+    the package may be using images from a directory outside that of the
+    plugin itself.
+    """
+
+    MONOSPACE_FONT_SIZE_DELTA: int = 0
+    """
+    The monospace (code) font size delta relative to Spyder default, in points.
+    """
+
+    INTERFACE_FONT_SIZE_DELTA: int = 0
+    """
+    The interface (UI) font size delta relative to Spyder default, in points.
+    """
+
+    CONTEXT_NAME: str | None = None
+    """
+    The name under which to store actions, toolbars, toolbuttons and menus.
+    """
+
+    CAN_BE_DISABLED: bool = True
+    """
+    Define if a plugin can be disabled in the Spyder :guilabel:`Preferences`.
+
+    If ``False``, the plugin is considered "core" and cannot be disabled.
+    ``True`` by default, meaning the plugin can be disabled.
+    """
+
+    REQUIRE_WEB_WIDGETS: bool = False
+    """
+    Declare whether the plugin needs to use Qt web widgets (QtWebEngine).
+
+    Qt Web Widgets is a heavy dependency for many packagers, e.g. Conda-Forge.
+    We thus ask plugins to declare whether or not they need
+    web widgets to make it easier to distribute Spyder without them.
+
+    `More information <https://github.com/spyder-ide/spyder/pull/22196#issuecomment-2189377043>`__
+    """
 
     # --- API: Signals -------------------------------------------------------
     # ------------------------------------------------------------------------
     # Signals here are automatically connected by the Spyder main window and
     # connected to the the respective global actions defined on it.
     sig_free_memory_requested = Signal()
     """
-    This signal can be emitted to request the main application to garbage
-    collect deleted objects.
+    Request the main application garbage collect deleted objects.
```

Comment:
This and all other signals below are not shown to be Qt signals in the rendered docs, so the docstrings for them don't make sense in the way you rewrote them.

So, please either add typing to these signals or mention that they are signals in their docstrings.